pcf/core/quasiparticle.py

```diff
@@ -164,9 +164,13 @@ def set_desired_state(self, desired_state):
 
         Args:
 
-            desired_state (State):
+            desired_state (str): one of running,stopped,terminated. Can also pass in the state directly. ex. State.running
         """
-        self.desired_state = desired_state
+        if isinstance(desired_state, str):
+            self.desired_state = STATE_STRING_TO_ENUM.get(desired_state)
```

Comment:
Need to add a .lower() to the `desired_state` in order to prevent errors. We also have the opportunity to throw a more specific error in here if the state can't be parsed, now that we are using strings rather than the actual State class.